src/api/EpiData.ts

```diff
@@ -142,18 +143,42 @@ export function loadDataSet(
   url.searchParams.set('format', 'json');
   return fetchImpl<Record<string, unknown>[]>(url)
     .then((res) => {
-      const data = loadEpidata(title, res, columns, columnRenamings, { _endpoint: endpoint, ...params });
-      if (data.datasets.length == 0) {
+      try {
+        const data = loadEpidata(title, res, columns, columnRenamings, { _endpoint: endpoint, ...params });
+        if (data.datasets.length == 0) {
+          return UIkit.modal
+            .alert(
+              `
+        <div class="uk-alert uk-alert-error">
+          <a href="${url.href}">API Link</a> returned no data.
+        </div>`,
+            )
+            .then(() => null);
+        }
+        return data;
+      } catch (error) {
+        console.warn('failed loading data', error);
+        // EpiData API error - JSON with "message" property
+        if ('message' in res) {
+          return UIkit.modal
+            .alert(
+              `
+          <div class="uk-alert uk-alert-error">
+            Failed to fetch API data from <a href="${url.href}">API Link</a>:<br/><i>${res['message']}</i>
+          </div>`,
+            )
+            .then(() => null);
+        }
+        // Fallback for generic error
         return UIkit.modal
           .alert(
             `
         <div class="uk-alert uk-alert-error">
-          <a href="${url.href}">API Link</a> returned no data.
+          Failed to fetch API data from <a href="${url.href}">API Link</a>.
```

Comment:
can we add more information to this, like the returned status code at the very least?  when a user runs into the rate limiting, this will be helpful.  in fact, when a 429 is served via the API, it returns this html in the body, which would be even more useful to a user:
```html
<!doctype html>
<html lang=en>
<title>429 Too Many Requests</title>
<h1>Too Many Requests</h1>
<p>Rate limit exceeded for anonymous queries. To remove this limit, register a free API key at https://api.delphi.cmu.edu/epidata/admin/registration_form</p>
```

Reply:
Also fixed - it should now include the error code if it's not inside the JSON.

Reply:
this isnt working for me when it encounters a 429 from making too many requests...  it still just says "Failed to fetch API data from API Link."